src/librustdoc/clean/mod.rs

```diff
@@ -366,6 +368,7 @@ pub struct Item {
     pub def_id: DefId,
     pub stability: Option<Stability>,
     pub deprecation: Option<Deprecation>,
+    pub non_exhaustive: bool,
```

Comment:
I think i'd prefer seeing the `non_exhaustive` marker just on the kinds where it's relevant, so just `Struct`/`Enum`/etc, instead of on `Item` here where literally everything has to care about it.

On the other hand, if i remember right we do keep all the `ast::Attribute`s around in the `Attributes` struct, so we could just query them as needed, rather than passing this flag around all the way from `clean_ast` to `html/render`.

Reply:
This should be resolved now.